### ISSUE TITLE
Printing MethodError for `LinearAlgebra.mul!` takes ~30 seconds

### ISSUE THREAD

**Issue by marius311** [JuliaLang/julia#30185]
On my laptop and Julia 1.0.2, the following takes about 30 seconds to complete, every single time (not just the first time):

```
julia> LinearAlgebra.mul!(1,2,3)
ERROR: MethodError: no method matching mul!(::Int64, ::Int64, ::Int64)
Closest candidates are:
  mul!(::UpperTriangular, ::Number, ::UpperTriangular) at /home/marius/src/julia/1.0.0/usr/share/julia/stdlib/v1.0/LinearAlgebra/src/triangular.jl:411
  mul!(::UpperTriangular, ::Number, ::UnitUpperTriangular) at /home/marius/src/julia/1.0.0/usr/share/julia/stdlib/v1.0/LinearAlgebra/src/triangular.jl:430
  mul!(::LowerTriangular, ::Number, ::LowerTriangular) at /home/marius/src/julia/1.0.0/usr/share/julia/stdlib/v1.0/LinearAlgebra/src/triangular.jl:449
  ...
Stacktrace:
 [1] top-level scope at none:0
```

Not sure what is special about `mul!`, other functions with just as many methods or more e.g. `+("a","b","c")` return the MethodError instantly.

**Comment by KristofferC** [comment]
FWIW, what is taking time is showing the method candidates and specifically, the type intersection at 

https://github.com/JuliaLang/julia/blob/a50d5af9cb9a7cce9a5ee3154bba735ffc74c1b1/base/errorshow.jl#L353-L354

**Comment by marius311** [comment]
Looks to have been fixed at some point, its near-instant now on 1.3.1 for me.
